ui/components/app/modals/contract-details-modal/contract-details-modal.js

```diff
@@ -43,6 +49,48 @@ export default function ContractDetailsModal({
   const addressBookEntry = useSelector((state) => ({
     data: getAddressBookEntry(state, toAddress),
   }));
+  const nft =
+    assetStandard === ERC721 ||
+    assetStandard === ERC1155 ||
+    // if we don't have an asset standard but we do have either both an assetname and a tokenID or both a tokenName and tokenId we assume its an NFT
+    (assetName && tokenId) ||
+    (tokenName && tokenId);
+
+  const tokenList = useSelector(getTokenList);
+
+  const nftTokenListImage = tokenList[tokenAddress.toLowerCase()]?.iconUrl;
+
+  let nftCollectionNameExist;
+  let nftCollectionImageExist;
+
+  Object.values(collections).forEach((nftCollections) => {
```

Comment:
I commented on some similar code in another PR here: https://github.com/MetaMask/metamask-extension/pull/15844#discussion_r999731505

If we need the same code here, it might be best to create a reusable component for rendering nft collection images